### ISSUE TITLE
Uploads not working - "Reserved field of websocket was not zero"

### ISSUE THREAD

**Issue by joeldeteves** [Ylianst/MeshCentral#4907]
**Describe the bug**
I am unable to upload files to any of my agents. The upload gets stuck on the progress bar. No errors are shown in the server logs or tracelog, however I see the following error being spammed in the agent `jlog`:

`Reserved field of websocket was not zero`

Everything else seems to work except this one feature. I can also download files no problem.

**To Reproduce**
Connect to an agent and try to upload a file to it

**Expected behavior**
Files should be uploaded to the agent

**Screenshots**
Here's a screenshot of the progress bar. This is a 58 KB file and it's been sitting here for 20 mins:
![image](https://user-images.githubusercontent.com/33190570/211924781-39d33db8-3000-40cb-a53b-1684b65fa5ca.png)

**Server Software (please complete the following information):**
 - OS: [e.g. Ubuntu]
 - Virtualization: [e.g. Docker] Kubernetes
 - Network: [e.g. LAN/WAN, reverse proxy, cloudflare, ssl offload, etc...] Traefik
 - Version: [e.g. 1.0.43] 1.1.2
 - Node: [e.g. 18.4.0] Not sure

**Client Device (please complete the following information):**
 - Device: [e.g. Laptop] Laptop
 - OS: [e.g. Ubuntu] Fedora 37
 - Network: [e.g. Local to Meshcentral, Remote over WAN] Remote over WAN
 - Browser: [e.g. Google Chrome] Same problem in Chrome + Firefox
 - MeshCentralRouter Version: [if applicable] n/a

**Remote Device (please complete the following information):**
 - Device: [e.g. Laptop] Desktop
 - OS: [e.g. Windows 10 21H2] Windows 10 21H2
 - Network: [e.g. Local to Meshcentral, Remote over WAN] Remote over WAN
 - Current Core Version (if known): [**HINT**: Go to a device then `console` Tab then type `info`] Dec 9 2022, 3593025475

**Your config.json file**
```
    {
      "settings": {
        "mongodb": "${encoded_uri}",
        "cert": "${MESH_HOST}",
        "tlsOffload": "${NGINX_HOST_IP}",
        "redirPort": 8080,
        "WANonly": true,
        "minify": 1,
        "port": 4443,
        "agentAliasPort": 443,
        "aliasPort": 443,
        "allowLoginToken": true,
        "allowFraming": true,
        "_agentPing": 60,
        "agentPong": 300,
        "allowHighQualityDesktop": true,
        "agentCoreDump": false,
        "compression": true,
        "wsCompression": true,
        "agentWsCompression": true,
        "maxInvalidLogin": {
          "time": 5,
          "count": 5,
          "coolofftime": 30
        }
      },
      "domains": {
        "": {
          "title": "Mesh Central",
          "newAccounts": false,
          "mstsc": true,
          "geoLocation": true,
          "certUrl": "https://${NGINX_HOST_IP}:${NGINX_HOST_PORT}",
          "agentConfig": [ "webSocketMaskOverride=${WS_MASK_OVERRIDE}" ],
          "authStrategies": {
            "azure": {
              "callbackurl": "https://${MESH_HOST}/auth-azure-callback",
              "newAccounts": true,
              "clientId": "${AAD_CLIENT_ID}",
              "clientSecret": "${AAD_CLIENT_SECRET}",
              "tenantId": "${AAD_TENANT_ID}"
            }
          },
          "showPasswordLogin": false,
          "passwordRequirements": {
            "allowaccountreset": false
          }
        }
      },
      "smtp": {
        "host": "${SMTP_HOST}",
        "port": "${SMTP_PORT}",
        "from": "${SMTP_FROM}",
        "user": "${SMTP_USER}",
        "pass": "${SMTP_PASS}",
        "tls": "${SMTP_TLS}"
      }
    }
```

**Comment by joeldeteves** [comment]
Adding to this @Ylianst it seems that it works when I use `IngressRouteTCP`, however this is not ideal because it does not allow me to use the middlewares I need. Further, every feature works except file transfer which is odd (file downloads from the same agent work fine).

So something with the Traefik http router appears to be screwing up the File uploads.

Appreciate any help you can provide in troubleshooting this issue so I know where to look (server logs aren't showing anything, and neither is the trace option in the web console).

PS - I also tried the `WebsocketMaskOverride=1` trick however it doesn't make any difference (and shouldn't be needed anymore since IIRC Traefik now handles this automatically).

**Comment by joeldeteves** [comment]
I got to the bottom of this issue with help from @NiceGuyIT - the issue turned out to be related to compression.

It turns out, Traefik does not handle websockets compression very well - setting the following solved it for me:

```
        "compression": false,
        "wsCompression": false,
        "agentWsCompression": false,
```

I also tested by enabling compression in the Traefik proxy itself and this yielded a performance improvement. So I think if you are running Mesh behind NGINX or as a standalone, using Mesh Compression is good, but if you use Traefik, it's better to let the proxy handle it.

I will go ahead and close this issue.
